addons/event_booth_sale/models/sale_order_line.py

```diff
@@ -38,11 +38,25 @@ def _compute_event_booth_pending_ids(self):
 
     def _inverse_event_booth_pending_ids(self):
         for so_line in self:
+            existing_booths = so_line.event_booth_registration_ids.event_booth_id
+            selected_booths = so_line.event_booth_pending_ids
+
+            booths_to_create = selected_booths\
+                .filtered(lambda booth: booth not in existing_booths) if existing_booths else selected_booths
+            booths_to_remove = existing_booths\
+                .filtered(lambda booth: booth not in selected_booths) if existing_booths else None
+
+            if booths_to_remove:
+                self.env['event.booth.registration'].search([
+                    ('event_booth_id', 'in', booths_to_remove.ids),
+                    ('sale_order_line_id', '=', so_line.id),
+                ]).unlink()
```

Comment:
Perhaps we could do something like this here to reduce a bit the diff:
```suggestion
            booth_regs_to_remove = so_line.event_booth_registration_ids.filtered(
                lambda reg: reg.event_booth_id not in selected_booths)
            booth_regs_to_remove.unlink()
```